File: ARC106/A.cpp
#include <bits/stdc++.h>
using namespace std;

#define ll long long
#define repeat(i, x) for (register ll i = 0; i < x; i++)

ll myPow(int base, int e)
{
  ll res = 1;
  for (int i = 0; i < e; i++)
  {
    res *= base;
  }
  return res;
}

int main()
{
  ll N;
  cin >> N;
  for (ll i = 0;; i++)
  {
    ll diff = N - myPow(5, i);
    if (diff <= 0)
    {
      cout << -1 << endl;
      return 0;
    }
    for (ll j = 0;; j++)
    {
      ll tmpPow = myPow(3, j);
      if (diff == tmpPow)
      {
        cout << j << " " << i << endl;
        return 0;
      }
      else if (diff < tmpPow)
      {
        break;
      }
    }
  }
}

File: ARC106/A.py
N = int(input())

i = 1
j = 1

while True:
  diff = N - pow(5, i)
  if diff <= 0:
    print(-1)
    exit(0)
  while True:
    tmpPow = pow(3, j)
    if diff == tmpPow:
      print(j,i)
      exit(0)
    if diff < tmpPow:
      break
    j+=1
  j=1
  i+=1
